erts/etc/common/erlexec.c

```diff
@@ -2121,14 +2139,15 @@ initial_argv_massage(int *argc, char ***argv)
     if (vix == (*argc > 1 ? 2 : 0)) {
```

Comment:
I suggest that you remove the entire `if` statement. It is an optimization that probably does not save enough time to be worth it. Also, it is much less likely to actually trigger nowadays since most users have `ERL_AFLAGS=-kernel shell_history enabled` in their environment.

Removing the `if` statement will also fix [this bug](https://github.com/erlang/otp/pull/7639#pullrequestreview-1618117165) noticed by @jchristgit.

Reply:
Good suggestion.